packages/coreutils/src/uuid.ts

```diff
@@ -1,18 +1,14 @@
 // Copyright (c) Jupyter Development Team.
 // Distributed under the terms of the Modified BSD License.
+import {
+  UUID
+} from '@phosphor/coreutils';
+
 
 /**
- * Get a random hex string (not a formal UUID).
- *
- * @param length - The length of the hex string.
+ * wraps the phosphor function, for backwards compatibility
  */
 export
-function uuid(length: number=32): string {
-  let s = new Array<string>(length);
-  let hexDigits = '0123456789abcdef';
-  let nChars = hexDigits.length;
-  for (let i = 0; i < length; i++) {
-    s[i] = hexDigits.charAt(Math.floor(Math.random() * nChars));
-  }
-  return s.join('');
+function uuid(): string {
```

Comment:
I didn't realize the old function had a length argument. Wow, that is *definitely* non-conforming.

We should probably keep the length argument, but it's not clear to me what we should do with it. 

(a) We could ignore it (but that is basically changing the behavior, though perhaps not the worst sin in the world?). 

(b) We could try to use it by generating the appropriate number of uuids and truncating the last one to be the right length. 

(c) Or we could just bite the bullet and declare that the next release is another major release.

I think (a) is a bad option since it breaks semver, and we've tried to be serious about semver in this repo. I think (b) is annoying, but question its usefulness. So I think I'm +0 on (c), and -0 on (b), and -1 on (a).

@ian-r-rose, @afshin?

Reply:
I can only say, that every usage of the function in the packages was using the default value. You can see in the changelog. But I don't know if the packages are used for extensions etc. 

It may be possible to revert the changes in the packages, use the original uuid function as a wrapper around the UUID.uuid4() and truncate the result if the length is specified.

Reply:
> It may be possible to revert the changes in the packages, use the original uuid function as a wrapper around the UUID.uuid4() and truncate the result if the length is specified.

I would say still move to using phosphor internally. The concern about backwards compatibility is because this is a package that a lot of outside extensions probably use too, and because we've already declared this package as 1.0 stability. Your proposal is basically what I mention in (b), though we'd have to handle lengths of more than 32 characters, so we'd first probably concatenate together the appropriate uuid strings

Reply:
OK. I will wait for the further instructions. Meanwhile I will look around for another good-first-issue where I can help :)